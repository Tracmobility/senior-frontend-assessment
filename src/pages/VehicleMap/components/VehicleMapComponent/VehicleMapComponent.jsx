import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid'; 
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';
import L from 'leaflet';
import 'leaflet/dist/leaflet.css';
import styles from './VehicleMapComponent.module.scss';

const VehicleMapComponent = ({ mockData }) => {
  const vehicleData = mockData[1];
  const icon = L.icon({ iconUrl: '/images/marker-icon.png' });

  return (
    <div className={styles.leafletMapContainer}>
      <MapContainer
        center={[vehicleData[0].lat, vehicleData[0].lng]}
        zoom={14}
        style={{ width: '100%', height: '100%' }}
      >
        <TileLayer
          attribution='&copy; <a href="https://stadiamaps.com/">Stadia Maps</a>, &copy; <a href="https://openmaptiles.org/">OpenMapTiles</a> &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors'
          url='https://tiles.stadiamaps.com/tiles/alidade_smooth_dark/{z}/{x}/{y}{r}.png'
        />
        {
          vehicleData.map(dataPoint => {
            const point = [dataPoint.lat, dataPoint.lng]

            return (
              <Marker position={point} key={uuidv4()} icon={icon}>
                <Popup>
                  <span>Time: {dataPoint.timeStamp}</span>
                  <br/>
                  <span>LATITUDE: {dataPoint.lat}</span>
                  <br/>
                  <span>LONGITUDE: {dataPoint.lng}</span>
                </Popup>
              </Marker>
            )
          })
        }
      </MapContainer>
    </div>
  )
};

VehicleMapComponent.propTypes = {
  mockData: PropTypes.shape({
    1: PropTypes.arrayOf(
      PropTypes.shape({
        lng: PropTypes.number.isRequired,
        lat: PropTypes.number.isRequired,
        timeStamp: PropTypes.string.isRequired,
      }).isRequired,
    ).isRequired,
  }).isRequired,
};

export default VehicleMapComponent;

