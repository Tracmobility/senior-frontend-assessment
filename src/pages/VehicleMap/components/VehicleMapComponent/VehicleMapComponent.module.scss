.leafletMapContainer {
  display: block;
  width: 98%;
  height: 95%;
  overflow: hidden;
  border-radius: 4px;
}
