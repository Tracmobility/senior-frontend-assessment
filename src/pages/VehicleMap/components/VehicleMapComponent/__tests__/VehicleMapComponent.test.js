import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import VehicleMapComponent from '../VehicleMapComponent';

Enzyme.configure({ adapter: new Adapter() });

const mockData = {
  1: [
    {
        "lng": -0.1204129,
        "lat": 51.5176398,
        "timeStamp": "01/01/2021 12:23"
    },
    {
        "lng": -0.1229235,
        "lat": 51.5171991,
        "timeStamp": "01/01/2021 12:23"
    },
    {
        "lng": -0.123138,
        "lat": 51.5172926,
        "timeStamp": "01/01/2021 12:23"
    },
  ]
}

const vehicleMapComponent = (<VehicleMapComponent mockData={mockData} />);

describe('It matches the previous working version of the vehicle map component', () => {
  it('Matches the previous snapshot of the vehicle map component', () => {
    const tree = renderer.create(vehicleMapComponent);
    expect(tree).toMatchSnapshot();
  });
});