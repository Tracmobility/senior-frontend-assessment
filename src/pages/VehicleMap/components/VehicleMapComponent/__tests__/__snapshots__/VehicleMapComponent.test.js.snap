// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the previous working version of the vehicle map component Matches the previous snapshot of the vehicle map component 1`] = `
<div
  className="leafletMapContainer"
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
